controllers/managedocs_controller.go

```diff
@@ -255,3 +344,138 @@ func (r *ManagedOCSReconciler) updateComponents() v1.ComponentStatusMap {
 
 	return components
 }
+
+func (r *ManagedOCSReconciler) checkUninstallCondition() bool {
+	cmNamespaceName := types.NamespacedName{
+		Name:      r.DeleteConfigMapName,
+		Namespace: r.namespace,
+	}
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(r.ctx, cmNamespaceName, configmap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false
+		}
+		r.Log.Error(err, "Unable to get configMap")
+		return false
+	}
+	labels := configmap.Labels
+	_, ok := labels[r.DeleteConfigMapLabelKey]
+	return ok
+}
+
+func (r *ManagedOCSReconciler) removeOlmComponents() error {
+
+	r.Log.Info("Deleting subscription")
+	subscription := &operators.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.AddonSubscriptionName,
+			Namespace: r.namespace,
+		},
+	}
+	if err := r.deleteResource(r.Client, subscription, r.AddonSubscriptionName); err != nil {
+		return err
+	}
+	r.Log.Info("subscription removed successfully")
+
+	r.Log.Info("Deleting CSV")
+	csvList := &operators.ClusterServiceVersionList{}
+	err := r.Client.List(r.ctx, csvList)
+	if err != nil {
+		r.Log.Error(err, "Unable to list CSVs")
+		return err
+	}
+	for _, csv := range csvList.Items {
+		if strings.HasPrefix(csv.Name, deployerCsvPrefix) {
+			deployerCsvName := csv.Name
+			csv := &operators.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      deployerCsvName,
+					Namespace: r.namespace,
+				},
+			}
+			if err := r.deleteResource(r.Client, csv, deployerCsvName); err != nil {
+				return err
+			}
+		}
+	}
+	r.Log.Info("CSV removed successfully")
+
+	return nil
+}
+
+func (r *ManagedOCSReconciler) areComponentsReadyForUninstall() bool {
+	subComponent := r.managedOCS.Status.Components
+
+	if subComponent.StorageCluster.State != v1.ComponentReady {
+		return false
+	}
+	return true
+}
+
+func (r *ManagedOCSReconciler) deleteComponents() error {
+	r.Log.Info("Deleting storageCluster")
+	if err := r.deleteResource(r.Client, &ocsv1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{Name: storageClusterName, Namespace: r.namespace},
+	}, storageClusterName); err != nil {
```

Comment:
Let's not inline object creation inside a function call, it makes the code unreadable. 
Also why do we need to create a `storagecluster` at all, we already create an load one in `reconcilePhases`

Reply:
The storagecluster object is created in other block under reconcilePhases and thus it will be unavailable for executing the above statement. Have removed the inline object creation.

Reply:
Can we just pass it to this function or save it on the reconciler?

Reply:
ack